### ISSUE TITLE
TreeView is extremely slow on remove/clear operations

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jimradford/superputty#516]
```
What steps will reproduce the problem?
Search for sessions in a large session list (1000s) or reload sessions.

What is the expected output? What do you see instead?
Reload should be much faster.

What version of the product are you using? On what operating system?
Win7 64, superputty 1.4.0.5

Please provide any additional information below.
The winform treeview is very slow in profiling on remove and clear operations. 
I have attempted to wrap clear in beginupdate endupdate to no avail.
```

Original issue reported on code.google.com by `j...@gimo.co.uk` on 29 Apr 2015 at 3:40

**Comment by jimradford** [comment]
I never imagined having 1000s of sessions, I'll see about implementing a virtual treeview for the sessions list.

**Comment by waukeerules** [comment]
I'm having the same issue with SuperPutty.. I'm using version 1.4.0.5, windows 7. And I only have about 300 sessions on my end.

**Comment by jimradford** [comment]
@waukeerules can you give it a shot with this build: https://ci.appveyor.com/api/buildjobs/c5p29ikv73i3xca3/artifacts/SuperPuttyInstaller/bin/Debug/SuperPuttySetup.msi I want to see of @joeyhub 's commit a few days ago did anything to improve session loading

**Comment by waukeerules** [comment]
I used the build on my personal laptop and it seemed to work okay. I will install it on my main build tomorrow.

**Comment by waukeerules** [comment]
Just tried it on my computer w/ over 1000+ sessions. It works exactly as you'd expect it to work. Very fast, thank you.

**Comment by jimradford** [comment]
Tanks Tim! This fix is in master and will be included in 1.4.0.7
